### PR TITLE
Add models and api views

### PR DESCRIPTION
**Pull request by aylar23**

### CODE REVIEW

File: socium/user/urls.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    DeclineFollowRequestApiView, UserAPIList, UserAPIUpdate
+
+app_name = 'user'
+
+urlpatterns = [
+    path("register/", UserRegistrationAPIView.as_view(), name="create-user"),
+    path("login/", UserLoginAPIView.as_view(), name="login-user"),
+    path('token/refresh/', TokenRefreshView.as_view(), name='token_refresh'),
+    path('token/verify/', TokenVerifyView.as_view(), name='token_verify'),
+    path('token/verify/', TokenVerifyView.as_view(), name='token_verify'),
```
Comment (PermanAtayev) [inline socium/user/urls.py:14]: User login edenden soňra ol user-e tokeni berseň, soňam hemme login gerek bolan endpointlerde request bilen gelen token-yň dogrulygyny ýa-da dälligini barlasaň has gowy bolar. 

Şu ýerde gowja makala bar eken: https://levelup.gitconnected.com/how-to-secure-your-rest-api-using-jwt-a923ba4a497e

File: socium/user/urls.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    path('api/request/<receiver_id>', FollowRequestsApiView.as_view(), name='request'),
+    path('api/request_accept/<request_id>', AcceptFollowRequestApiView.as_view(), name='request-accept'),
+    path('api/request_decline/<request_id>', DeclineFollowRequestApiView.as_view(), name='request-decline'),
```
Comment (PermanAtayev) [inline socium/user/urls.py:17]: Rest api-de endpointerde işlik ulanmak maslahat berilenok. Her endpoint bir resource bilen baglanyşykly bolmaly

Follow we unfollow-y rest bilen şular ýaly gurup bilersiň: https://softwareengineering.stackexchange.com/questions/339751/options-for-the-restful-approach-for-follow-and-unfollowing

**Review by PermanAtayev — CHANGES_REQUESTED** [review]

File: socium/user/urls.py (hunk 1 above)
Comment (aylar23) [reply socium/user/urls.py:14]: Barlayana:
class FollowRequestsApiView(GenericAPIView):
    permission_classes = [IsAuthenticated]  user authenticatedmi dal-mi sony  barlayar

File: socium/user/permissions.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from rest_framework import permissions
+
+
+class IsOwnerOrReadOnly(permissions.BasePermission):
+
+    def has_object_permission(self, request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return obj == request.user
```
[the remark this answers is not in the document]
Comment (aylar23) [reply socium/user/permissions.py:9]: bardaki obj user, meselem user profile-ini uytgetjek bolsa dine ozuninkisini uytgedip biler yaly, basgasynynky bolsa dine safe method-leri ulanyp bilyar.

File: socium/user/models.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+REQUEST_STATUS = (
+    ('sent', 'sent'),
+    ('accepted', 'accepted')
+)
+
+
+class FollowRequest(models.Model):
+    sender = models.ForeignKey(User, on_delete=models.CASCADE, related_name='sender')
+    receiver = models.ForeignKey(User, on_delete=models.CASCADE, related_name='receiver')
+    updated = models.DateTimeField(auto_now=True)
```
[the remark this answers is not in the document]
Comment (aylar23) [reply socium/user/models.py:39]: bir zatlar etjek boldum, ayyrmak yadymdan cykypdyr

File: socium/user/urls.py (hunk 2 above)
Comment (aylar23) [reply socium/user/urls.py:17]: Done

File: socium/user/models.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    created = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return self.id
+
+    def accept(self):
+        self.sender.followers.add(self.receiver)
+        self.receiver.following.add(self.sender)
```
[the remark this answers is not in the document]
Comment (aylar23) [reply socium/user/models.py:47]: Hawa, duzetdim

**Pull request merged**